src/main/java/stream/flarebot/flarebot/util/GeneralUtils.java

```diff
@@ -72,4 +82,25 @@ public static String formatCommandPrefix(TextChannel channel, String usage) {
         String prefix = String.valueOf(get(channel));
         return usage.replaceAll("\\{%\\}", prefix);
     }
+
+    public static AudioItem resolveItem(Player player, String input) throws IllegalArgumentException, YoutubeAccessException {
+        Optional<AudioItem> item = Optional.empty();
+        boolean failed = false;
+        for (int i = 0; i <= 2; i++) {
+            try {
+                item = Optional.ofNullable(player.resolve(input));
+                failed = false;
+                break;
+            } catch (FriendlyException | InterruptedException | ExecutionException e) {
+                failed = true;
+            }
+        }
+        if (failed) {
+            throw new YoutubeAccessException();
```

Comment:
Just use another built in exception to be honest, it's really the youtube state being bad, so IllegalStateException would work

Also in the future there's a big chance it won't be only youtube